src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -325,7 +327,13 @@ export default class GeFormRenderer extends LightningElement{
         });
     }
 
+    handleWidgetStateChange(changeEvent) {
+        this._isElevateWidgetInDisabledState = (changeEvent.state === 'readOnly');
```

Comment:
I wouldn't consider this a blocker and we can keep this as-is.

I'm just a bit uneasy about "spilling" the tokenize widget's internal display state values to the form renderer. But if we keep this, this dispatch logic should be in the 'readOnly' onEnter method.

Reply:
Thanks for raising your concern here. I now get that ideally the widget's internal display state values should stay well, internal.

Ultimately I need a way for geFormRender to know when the widget is in a read-only state as we're using that for one of the requirements to determine whether or not we should tokenize. I believe there are mechanisms in place already that _should_ be doing this, but it did no appear to be working. Perhaps this is just a band-aid for now, but ideally - it'd be great to connect with you and figure out what the ideal path forward is. If this is a non-blocker, maybe we can look into getting this correctly developed in #6396 ?

Reply:
I'm with ya. The existing mechanism is tied to the "Do not enter payment" behavior.

The event name is "doNotChargeState" (bad name). Prior to the last round of changes in this component, that one event had dual responsibilities for setting the "isElevateWidgetDisabled" property in the renderer for the "Do not enter payment" user action and for when the widget went into read-only mode.

I think ideally we rename that event and additionally start firing it from the `DISPLAY_DEFINITIONS.readOnly.onEnter` and `DISPLAY_DEFINITIONS.readOnly.onExit` as needed.

Let me know what you think